src/common/os/posix/os_utils.cpp

```diff
@@ -393,4 +395,147 @@ void CtrlCHandler::handler(void*)
 	terminated = true;
 }
 
+namespace {
+
+	class Converter
+	{
+	public:
+		virtual char* convert(const char* fromBuf, size_t fromLen) = 0;
+		virtual void destroy(char* &outBuf) = 0;
+	};
+
+	// No real conversion, system locale is UTF-8 or iconv is not available
+	class NullConverter : public Converter
+	{
+		char* convert(const char* fromBuf, size_t fromLen) override
+		{
+			return const_cast<char*>(fromBuf);
+		}
+		void destroy(char* &outBuf)
+		{
+			outBuf = NULL;
+		}
+	};
+
+#ifdef HAVE_ICONV_H
+	class IConverter : public Converter
+	{
+	public:
+
+		IConverter(const char* sysLocale) : ic(NULL)
```

Comment:
On 06/29/2016 12:59 PM, Dimitry Sibiryakov wrote:
>> +		char* convert(const char* fromBuf, size_t fromLen) override
>> +		{
>> +			return const_cast<char*>(fromBuf);
>> +		}
>> +		void destroy(char* &outBuf)
>> +		{
>> +			outBuf = NULL;
>> +		}
>> +	};
>> +
>> +#ifdef HAVE_ICONV_H
>> +	class IConverter : public Converter
>> +	{
>> +	public:
>> +
>> +		IConverter(const char* sysLocale) : ic(NULL)
> They are not same. This one is consuming 4 times less memory, but do conversion from UTF-8
> only. Different name to be avoid Linux symbol hell.

In that case one class can be derived from another.
Anyway please avoid code duplication!